gstreamer-pbutils/src/auto/encoding_target.rs

```diff
@@ -0,0 +1,124 @@
+// This file was generated by gir (d933f9a+) from gir-files (???)
+// DO NOT EDIT
+
+use EncodingProfile;
+use Error;
+use ffi;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::mem;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct EncodingTarget(Object<ffi::GstEncodingTarget>);
+
+    match fn {
+        get_type => || ffi::gst_encoding_target_get_type(),
+    }
+}
+
+impl EncodingTarget {
+    pub fn new(name: &str, category: &str, description: &str, profiles: &[EncodingProfile]) -> EncodingTarget {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib_full(ffi::gst_encoding_target_new(name.to_glib_none().0, category.to_glib_none().0, description.to_glib_none().0, profiles.to_glib_none().0))
+        }
+    }
+
+    pub fn load<'a, P: Into<Option<&'a str>>>(name: &str, category: P) -> Result<EncodingTarget, Error> {
+        assert_initialized_main_thread!();
+        let category = category.into();
+        let category = category.to_glib_none();
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret = ffi::gst_encoding_target_load(name.to_glib_none().0, category.0, &mut error);
+            if error.is_null() { Ok(from_glib_full(ret)) } else { Err(from_glib_full(error)) }
+        }
+    }
+
+    pub fn load_from_file(filepath: &str) -> Result<EncodingTarget, Error> {
```

Comment:
Needs to be a &Path, can be fixed with a gir annotation in GStreamer. See gst_registry_add_path() or similar